### DISCUSSION TITLE
ftp not connected, bootloader kicking in?

### DISCUSSION THREAD

**Discussion by dionysius** [Freetz-NG/freetz-ng#911] [push firmware] [answered]
_Dürft gerne in Deutsch antworten. Answers in German are also welcome!_

This is my first time trying to flash a fritzbox. I have a ~7490~ 5490 (FRITZ!OS 07.31, firmware 151.07.31) and successfully built a image (standard selection - no modifications except model and version)

When I try to reboot (or pull the plug for 5secs) it always says it can't connect to ftp. I **am** connected directly with LAN. Currently the fritzbox is configured with 192.168.0.1/24 range and many services are turned off since I don't use them. Does anything need to be enabled for this to work? Or am I missing something else?

My attempts look like this:
```
$ ./tools/push_firmware -ip 192.168.0.1

 * Analyzing 'images/7490_07.57.all_freetz-ng-b2eb3cc50-devuan-support_20231219-050051.image' ... 
 * Product: Fritz_Box_HW185 (FRITZ!Box 7490)
 * Using command: ftp
 * Target host: 192.168.0.1
 * Outgoing IP: 192.168.0.80
 * Flash mode: ram-boot
 * Allowed memory size: 128 MB
 * Designated linux_fs_start: <other>

 !!! WARNING !!! WARNING !!! WARNING !!! WARNING !!! WARNING !!!
 !!!  THERE IS NO WARRANTY AT ALL !!! USE AT YOUR OWN RISK   !!!

 * Are you sure, that you want to flash this file to the device?
   images/7490_07.57.all_freetz-ng-b2eb3cc50-devuan-support_20231219-050051.image
   Proceed? (y/[n]) y

 * You should now reboot your box (192.168.0.1). Waiting for shut down.
   Switch off, if reboot is not detected because it happens too quickly.
   Some newer bootloader versions allow to flash on power-cycle only.
   .........................................................................................................................
 * MAPSTART=0x80000000
 * FULLSIZE=0x08000000	 (128 MB)
 * MAPLIMIT=0x88000000
 * FILESIZE=0x02468400	 (~36 MB)
 * ALIBYTES=0x00000000	 (~0 kB)
 * FREESIZE=0x05b97c00	 (~91 MB)
 * MTDSTART=0x85b97c00

 * No reply from box, assuming switch-off or restart. Trying to re-detect box.
   Waiting ............................................................. found!

ftp: Can't connect to `192.168.0.1:21': Connection refused
ftp: Can't connect to `192.168.0.1:ftp'
 * Detected linux_fs_start: '<unset>'
 * Designated linux_fs_start: 1

 * No reply from box, assuming switch-off or restart. Trying to re-detect box.
   Waiting . found!

 * Box is back up again, initiating transfer.

ftp: Can't connect to `192.168.0.1:21': Connection refused
ftp: Can't connect to `192.168.0.1:ftp'
Not connected.
Debugging on (debug=1).
Not connected.
Not connected.
Not connected.
Not connected.
Not connected.
Not connected.

done
```

Looking at similiar discussions led me to https://github.com/Freetz-NG/freetz-ng/discussions/641, but I had no success forcing it.
I tried `./tools/push_firmware -ip 192.168.0.1 -w -f` with a reboot or pulling the plug for 5s and land to the same issue as logged above. Is it a timing issue or something else?

**Comment by aus-sua** [comment, marked as answer]
> ./tools/push_firmware -ip 192.168.0.1

Wenn Du Deine 7490er nicht schon komplett "verbastelst" hast, ist der
bootloader-ftp einer/jeder FRITZ!Box immer unter 192.168.178.1 erreichbar, 
unabhängig (!) von "Deiner" festgelegten IP der 7490er.

Du musst also zum pushen ein IP-range von z.B. 192.168.178.1/24 verwenden
(und ggf. auch die IP des push-PCs darin fest vergeben).

Oder eben einfach nur ein:

./tools/push_firmware

dann übernimmt push_firmware das für Dich. :-)

**Reply by dionysius** [reply]
Great, this was an important detail I missed! Also: The linux machine needs to be set to 192.168.178.X even when link is down - so if it comes up, it is ready to connect.

That means I usually ran `./tools/push_firmware -w` and power-cycled manually because I was sure the pushing machine was ready.

Off-topic: I accidentally loaded a 7490 image, **but I have a 5490**!!! Don't ask me how I mixed those up. After a long recovery process it is fixed.
